kibana.md

```diff
@@ -1101,3 +1101,201 @@ an admin, who has just configured access for a new user. They would like
+   
+  if (metadata.customMetadata && metadata.customMetadata.rejectBasicAuth && basicHeaderAuth) {
+     return res.status(401).json({ message: 'Machine to machine communication is not allowed' });
+  }
+
+  return next()
+}
+```
+You can pass any custom metadata and based on it accepts or reject the specific request
+
+**⚠️IMPORTANT** Custom middleware must return `next()` function, to not block the request
+
+### Available rorRequest API
+You can access the rorRequest API via `req.rorRequest` in your custom middleware. The available options are:
+
+| Property name                                                          | Return value type            | Example return value                                                         | Description                                                                    |
+|:-----------------------------------------------------------------------|------------------------------|------------------------------------------------------------------------------|--------------------------------------------------------------------------------|
+| getAuthorizationHeaders()                                              | Map<string, string>          | Map(2) {'authorization' => 'Basic BWRtaW46ZGV2', 'cookie' => 'cookie value'} | Get headers using in the authorization                                         |
+| getCookies()                                                           | string or undefined          | 'cookie value'                                                               | Get the request cookies                                                        |
+| getMethod()                                                            | Method                       | 'GET'                                                                        | Get the request method                                                         |
+| getPath()                                                              | string                       | '/s/default/internal/bsearch'                                                | Get the request path                                                           |
+| getUrl()                                                               | string                       | '/s/default/internal/bsearch?compress=true'                                  | Get the request URL                                                            |
+| getBody()                                                              | Body or null or undefined    | { id: 42 }                                                                   | Get the request body                                                           |
+| getParams()                                                            | ParamsDictionary             | { id: '1234' }                                                               | Get URL parameters                                                             |
```

Comment:
Why don't we document only the `rorRequest`methods that are not already evident in the standard Express.js `req` object?

No much value to let them use `getPath`,`getUrl`, `getParams`, `getQueries`, `getHeaders`, `getOriginAddress`, `getMethod`, `getCookies`, ...  when they already have `req`, maybe we can put note about this and add a link to Express.js `req` type definition instead?

Reply:
Right, i removed redundant information and add info about express req and res objects